builders/interact/substrate-api/chopsticks.md

```diff
@@ -0,0 +1,203 @@
+
+```bash
+yarn start run-block  \
+--endpoint wss://fraa-dancebox-3001-rpc.a.dancebox.tanssi.network  \
+--output-path=./chain-output.json  \
+--block 1000
+```
+
+## WebSocket Commands {: #websocket-commands }
+
+Chopstick's internal WebSocket server has special endpoints that allow the manipulation of the local Substrate chain. 
+
+These are the methods that can be invoked and their parameters:
+
+|      Method      |      Parameters       |                          Description                          |
+|:----------------:|:---------------------:|:-------------------------------------------------------------:|
+|  `dev_newBlock`  |       `options`       |               Generates one or more new blocks.               |
+| `dev_setStorage` | `values`, `blockHash` |         Create or overwrite the value of any storage.         |
+| `dev_timeTravel` |        `date`         |     Sets the timestamp of the block to the `date` value.      |
+|  `dev_setHead`   |    `hashOrNumber`     | Sets the head of the blockchain to a specific hash or number. |
+
+??? code "Parameters details"
+
+    |   Parameter       |               Format                |                                Example                                 |
+    |:-----------------:|:-----------------------------------:|:----------------------------------------------------------------------:|
+    |   `options`       | `{ "to": number, "count": number }` |                            `{ "count": 5 }`                            |
+    |    `values`       |              `Object`               | `{ "Sudo": { "Key": "0x6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b" } }`  |
+    |  `blockHash`      |              `string`               | `"0x1a34506b33e918a0106b100db027425a83681e2332fe311ee99d6156d2a91697"` |
+    |     `date`        |               `Date`                |                        `"2030-08-15T00:00:00"`                         |
+    | `hashOrNumber`    |         `number | string`           |                                                                        |
+
+    - **`options` { "to": number, "count": number }** - a JSON object where `"to"` will create blocks up to a certain value, and `"count"` will increase by a certain number of blocks. Use only one entry at a time within the JSON object  
+    - **`values` Object** - a JSON object resembling the path to a storage value, similar to what you would retrieve via Polkadot.js  
+    - **`blockHash` string** - optional, the block hash at which the storage value is changed  
+    - **`date` Date** - a Date string (compatible with the JavaScript Date library) that will change the timestamp at which the next blocks being created will be. All future blocks will be sequentially created after that point in time  
+    - **`hashOrNumber` number | string** - if found, the chain head will be set to the block with the block number or block hash of this value  
```

Comment:
I think what I don't like about this is that we have the parameter information for the same parameter in two different spots. Like we have the parameter format and example in the table then below the table there is a description for that same parameter.

I would say let's just add a description column to the table so that the parameter details are all in one place, but I think that would make the table look too crowded and it wouldn't be the most readable.

So with all of this in mind, and the fact that these parameters are reused across the different methods, I think we should do something like this:

```
- `dev_newBlock` - generates one or more new blocks

     === "Parameters"
         
         - `options` -  a JSON object where `"to"` will create blocks up to a certain value, and `"count"` will increase by a certain number of blocks. Use only one entry at a time within the JSON object. This can be formatted as follows: `{ "to": number, "count": number }`
         
     === "Example"
     
         ```js
         import { WsProvider } from '@polkadot/api'
         const ws = new WsProvider(`ws://localhost:8000`)
         await ws.send('dev_newBlock', [{ count: 5 }])
         ```
```

The example is probably not formatted correctly to be able to copy/paste in a terminal, because I just wanted to get my general idea across. But I think if we can clean it up like this, it will look a lot nicer and be easier to read.

If you have questions, just ping me in slack 🙂

Reply:
Marking it as solved, even though it's wip in another branch.